compiler/qsc_project/src/manifest.rs

```diff
@@ -0,0 +1,87 @@
+use crate::Error;
+use serde::Deserialize;
+use std::path::PathBuf;
+use std::{
+    env::current_dir,
+    fs::{self, DirEntry, FileType},
+};
+
+pub const MANIFEST_FILE_NAME: &str = "qsharp.json";
+
+/// A Q# manifest, used to describe project metadata.
+#[derive(Deserialize, Debug, Default)]
+pub struct Manifest {
+    pub author: Option<String>,
+    pub license: Option<String>,
+    #[serde(default)]
+    pub exclude_files: Vec<String>,
+}
+
+/// Describes the contents and location of a Q# manifest file.
+#[derive(Debug)]
+pub struct ManifestDescriptor {
+    pub manifest: Manifest,
+    pub manifest_dir: PathBuf,
+}
+
+impl ManifestDescriptor {
+    pub(crate) fn new(manifest: Manifest, manifest_dir: PathBuf) -> Self {
+        Self {
+            manifest,
+            manifest_dir,
+        }
+    }
+}
```

Comment:
nitpick: this may just be a style thing, but I don't see the need for a `new` constructor when the fields are all public and the only place it is constructed is inside the crate. The one call location could just create the struct directly, which is what any callers outside the crate would do since the struct is pub and `new` is only pub(crate).

Reply:
I don't feel strongly about it, so it's fine to leave as-is. I think reading through the code I was surprised to see a constructor that "wasn't doing anything" paired with public fields, but it's not something I feel strongly enough to advocate one way or the other.